codegen/src/lib.rs

```diff
@@ -95,6 +95,9 @@
 #![feature(i128_type)]
 #![allow(unused_attributes)]
 #![allow(deprecated)]
+#![cfg_attr(feature="clippy", feature(plugin))]
+#![cfg_attr(feature="clippy", plugin(clippy))]
+#![cfg_attr(feature="clippy", allow(needless_lifetimes))]
```

Comment:
The reason for allowing this lint is because it was producing many false positives on Rocket's code base (tracked by https://github.com/Manishearth/rust-clippy/issues/740)